consensus/src/simplex/actors/batcher/actor.rs

```diff
@@ -439,65 +439,71 @@ impl<
+                    if let Some(signer) = self.participants.key(invalid) {
+                        warn!(?signer, "blocking peer for invalid signature");
+                        self.blocker.block(signer.clone()).await;
+                    }
+                }
+
+                // Store verified votes for certificate construction
+                let round = work.get_mut(&view).expect("round must exist");
+                for valid in voters {
+                    round.add_verified(valid);
+                }
+            } else {
                 timer.cancel();
                 trace!(
                     %current,
                     %finalized,
                     waiting = work.len(),
                     "no verifier ready"
                 );
-                continue;
-            };
-            timer.observe();
-
-            // Process verified votes
-            let batch = voters.len() + failed.len();
-            trace!(%view, batch, "batch verified votes");
-            self.verified.inc_by(batch as u64);
-            self.batch_size.observe(batch as f64);
-
-            // Block invalid signers
-            for invalid in failed {
-                if let Some(signer) = self.participants.key(invalid) {
-                    warn!(?signer, "blocking peer for invalid signature");
-                    self.blocker.block(signer.clone()).await;
-                }
             }
 
-            // Store verified votes for certificate construction
-            let round = work.get_mut(&view).expect("round must exist");
-            for valid in voters {
-                round.add_verified(valid);
-            }
-
-            // Try to construct and forward certificates
-            if let Some(notarization) = self
-                .recover_latency
-                .time_some(|| round.try_construct_notarization(&self.scheme))
-            {
-                debug!(%view, "constructed notarization, forwarding to voter");
-                voter
-                    .recovered(Certificate::Notarization(notarization))
-                    .await;
-            }
-            if let Some(nullification) = self
-                .recover_latency
-                .time_some(|| round.try_construct_nullification(&self.scheme))
-            {
-                debug!(%view, "constructed nullification, forwarding to voter");
-                voter
-                    .recovered(Certificate::Nullification(nullification))
-                    .await;
-            }
-            if let Some(finalization) = self
-                .recover_latency
-                .time_some(|| round.try_construct_finalization(&self.scheme))
-            {
-                debug!(%view, "constructed finalization, forwarding to voter");
-                voter
-                    .recovered(Certificate::Finalization(finalization))
-                    .await;
+            // Try to construct and forward certificates for the current view.
```

Comment:
nit:
```
            // Try to construct and forward certificates for the current view.
            // This runs regardless of whether batch verification happened above,
            // because the last vote processed may have come from us.
```